qiskit_nature/second_q/operators/polynomial_tensor.py

```diff
@@ -0,0 +1,126 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Polynomial Tensor class"""
+
+from __future__ import annotations
+from typing import Dict, Mapping
+from numbers import Number
+import numpy as np
+from qiskit.quantum_info.operators.mixins import (
+    LinearMixin,
+    AdjointMixin,
+    TolerancesMixin,
+)
+
+
+class PolynomialTensor(LinearMixin, AdjointMixin, TolerancesMixin):
+    """Polynomial Tensor class"""
+
+    def __init__(self, data: Mapping[str, np.ndarray | Number]):
+        copy_dict: Dict[str, np.ndarray] = {}
+
+        shapes = set()
+        for key, value in data.items():
+            if isinstance(value, Number):
+                value = np.asarray(value)
+
+            if len(value.shape) != len(key):
+                raise ValueError(
+                    f"Data key {key} of length {len(key)} does not match "
+                    f"data value matrix of dimensions {value.shape}"
+                )
+
+            dims = set(value.shape)
+
+            if len(dims) > 1:
+                raise ValueError(
+                    f"For key {key}: dimensions of value matrix are not identical {value.shape}"
+                )
+
+            shapes.update(dims)
+            copy_dict[key] = value
+
+        if len(shapes) != 1:
+            raise ValueError("Dimensions of value matrices in data dictionary are not identical.")
+
+        self._data = copy_dict
+
+    def mul(self, other: complex):
+        """Scalar multiplication of PolynomialTensor with complex"""
+
+        if not isinstance(other, Number):
+            raise TypeError(f"other {other} must be a number")
+
+        prod_dict: Dict[str, np.ndarray] = {}
+        for key, matrix in self._data.items():
+            prod_dict[key] = np.multiply(matrix, other)
+        return PolynomialTensor(prod_dict)
+
+    def _multiply(self, other):
+        return self.mul(other)
```

Comment:
```suggestion
```
Simply implementing `_multiply` above is fine, as the `MultiplyMixin` deals with exposing this method to the public API

Reply:
I did this to avoid unit testing a private method. But I can look for a way around it.

Reply:
You can still unittest the `mul` method (or actually just use `number * tensor`)

Reply:
Yes, was just looking into this. Will make the changes soon.